src/nb_func.cpp

```diff
@@ -364,15 +364,16 @@ PyObject *nb_func_new(const void *in_) noexcept {
     func->vectorcall = func->complex_call ? nb_func_vectorcall_complex
                                           : nb_func_vectorcall_simple;
 
+#if !defined(NB_FREE_THREADED)
     // Register the function
-    nb_ptr_map::iterator it;
     {
-        lock_internals guard(internals_);
+        nb_ptr_map::iterator it;
         bool success;
         std::tie(it, success) = internals_->funcs.try_emplace(func, nullptr);
```

Comment:
Since you no longer need the additional scope for the lock guard's lifetime, I think you could switch this back to the `auto [it, success] = ...` at top-level that it was before the FT changes.